tools/gulp-extras.js

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+"use strict";
+
+var fs = require("fs");
+var gutil = require("gulp-util");
+var path = require("path");
+var through = require("through2");
+
+/**
+ * Pretty logger using gutil.log
+ * @param {string} pluginName Name of the pluginName
+ * @param {Object} file A gulp file to report on
+ * @param {string} message The error message to display
+ */
+var logError = function(pluginName, file, message) {
+    var sourcePath = path.relative(__dirname, file.path).replace("../","");
+    gutil.log(`[${gutil.colors.cyan(pluginName)}] ${gutil.colors.red("error")} ${sourcePath}: ${message}`);
+};
+
+/**
+ * Plugin to verify the Microsoft copyright notice is present
+ */
+var checkCopyright = function() {
+    var re = /\/\/ Copyright \(c\) Microsoft Corporation. All rights reserved.\s+\/\/ Licensed under the MIT license. See LICENSE file in the project root for details.\s+/;
```

Comment:
Is it okay if we have the copyright in the middle of the file?
I also think that we probably don't want arbitrary spaces to appear after the first line...